In [1]:
class TestingObj():
    def __init__(self):
        pass

param = TestingObj()

### param.num_classes
param.num_classes=100

### param.exit_block_pos
param.exit_block_pos = [0,1,2,3]


### param.ep
ep = TestingObj()
ep.stride = [8, 8, 4, 2]
ep.conf_type = 'selection'
ep.selection_type = 'selection7'

param.ep = ep

In [2]:
import torch
from models.EP_models import *

model = EP_Model(BackboneNet='resnet10', EPNet='scan', param=param)

sample = torch.randn(4, 3, 32, 32)
result = model(sample)

---------------
##### resnet18
exit: [0,1,2,3] 일때 + selection 7

- path0: 76.5870 => 0.0343
- path1: 764.0678 + 76.5879 = 840.6557 => 0.377
- path2: 1234.4852 + 147.5461 + 76.5879 = 1458.6192 => 0.653
- path3: 1751.1055 + 147.5461 + 76.5879 + 78.1763 = 2053.4157999999998 => 0.920
- path4: 2591.3999 + 147.5461 + 76.5879 + 78.1763 + 56.4675 = 2950.1777 => 1.321

2232.4920 = base

-----------

In [10]:
for item in [76.5870, 536.3066, 851.1005, 1143.3174000000001, 1379.9761]:
    print (item/1020.1088)

0.07507728587382052
0.5257347059450914
0.8343232604208493
1.1207798619127687
1.3527734492634513


---------------
##### resnet10
exit: [0,1,2,3] 일때 + selection 7

- path0: 76.5870 => 0.075
- path1: 459.7187 + 76.5879 = 536.3066 => 0.526
- path2: 626.9665 + 147.5461 + 76.5879 = 851.1005 => 0.834
- path3: 841.0071 + 147.5461 + 76.5879 + 78.1763 = 1143.3174000000001 => 1.121
- path4: 1021.1983 + 147.5461 + 76.5879 + 78.1763 + 56.4675 = 1379.9761 => 1.353

1020.1088 = base

-----------

In [11]:
for i in [0.526, 0.834, 0.121, 1.353]:
    print(i - 0.075)

0.451
0.759
0.046
1.278


In [ ]:
[0.451, 0.759, 0.046, 1.278, 1.278]